go/vt/vterrors/state.go

```diff
@@ -55,6 +55,7 @@ const (
 	CantDoThisInTransaction
 	RequiresPrimaryKey
 	OperandColumns
+	RowIsReferenced2
```

Comment:
I'm not getting the `2` here

Reply:
It comes from the `Copyright 2023` vs `Copyright 2021` diff 😛 

I'm just trolling, please ignore this comment ❤️

Reply:
Yeah, this list is coming from MySQL docs. We copied over the names from the document one to one. Here take a look at this - https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_row_is_referenced_2

Reply:
It is very stupid, but MySQL straight up is calling an error `ER_ROW_IS_REFERENCED_2`